File: sketches/java/HelloSpring/src/main/java/FizzBuzzGenerator.java
import java.util.List;


public class FizzBuzzGenerator {

	List<INumberFormatter> _formatters;
	
	public FizzBuzzGenerator(List<INumberFormatter> formatters) {
		_formatters = formatters;
	}
	
	String get_output(int maximum) {
		
		String result = new String();
		for(int i=1; i<=maximum; i++)
			result += get_output_for(i) + " ";
		return result;
	}
	
	String get_output_for(int number) {
		
		for(INumberFormatter formatter : _formatters) {
			if( formatter.can_handle(number))
				return formatter.handle(number);
		}
		
		return null;
	}
	
}

File: sketches/spring/HelloSpring/HelloSpring/FizzBuzzGenerator.cs
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HelloSpring
{
    public class FizzBuzzGenerator
    {
        List<INumberFormatter> _formatters;

        public FizzBuzzGenerator(List<INumberFormatter> formatters)
        {
            _formatters = formatters;
        }

        public string GetOutput(int maximum)
        {
            var result = new StringBuilder();
            for (var i = 1; i <= maximum; i++)
                result.Append(GetOutputFor(i)).Append(" ");
            return result.ToString();
        }

        string GetOutputFor(int number)
        {
            return (from numberFormatter in _formatters
                    where numberFormatter.CanHandle(number)
                    select numberFormatter.Handle(number)).FirstOrDefault();
        }
    }
}